src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -367,51 +366,72 @@
-
-     (comidi/GET ["/catalog/" [#".*" :rest]] request
-                 (request-handler (assoc request :include-code-id? true)))
-     (comidi/POST ["/catalog/" [#".*" :rest]] request
-                  (request-handler (assoc request :include-code-id? true)))
-     (comidi/PUT ["/report/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/resource_type/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/resource_types/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/environment/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET "/environments" request
-                 (request-handler request))
-     (comidi/GET ["/status/" [#".*" :rest]] request
-                 (request-handler request))
+(schema/defn ^:always-validate
+  v3-ruby-routes :- bidi-schema/RoutePair
+  "v3 route tree for the ruby side of the master service."
+  [request-handler :- IFn]
+  (comidi/routes
+   (comidi/GET ["/node/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_content/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_metadatas/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_metadata/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_bucket_file/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/PUT ["/file_bucket_file/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/HEAD ["/file_bucket_file/" [#".*" :rest]] request
+                (request-handler request))
+
+   (comidi/GET ["/catalog/" [#".*" :rest]] request
+               (request-handler (assoc request :include-code-id? true)))
+   (comidi/POST ["/catalog/" [#".*" :rest]] request
+                (request-handler (assoc request :include-code-id? true)))
+   (comidi/PUT ["/report/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/resource_type/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/resource_types/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/environment/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET "/environments" request
+               (request-handler request))
+   (comidi/GET ["/status/" [#".*" :rest]] request
+               (request-handler request))))
 
+(schema/defn ^:always-validate
+  v3-clojure-routes :- bidi-schema/RoutePair
+  "v3 route tree for the clojure side of the master service."
+  [clojure-request-wrapper :- IFn
+   jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
+   get-code-content-fn :- IFn]
+  (let [environment-class-handler
+        (clojure-request-wrapper (environment-class-handler jruby-service))
```

Comment:
I think the way @cprice404 was envisioning this would work is that the whole route subtree would somehow be wrapped by the `clojure-request-wrapper` rather than having to individually wrap each sub-handler with the `clojure-request-wrapper`.  TBH, though, I'm not sure if wrapping the whole sub-tree is something you can do with comidi/bidi.

@cprice404 - can you provide a sanity check here?